cyclonedx-bom/src/models/composition.rs

```diff
@@ -101,6 +101,21 @@ impl AggregateType {
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct BomReference(pub(crate) String);
 
+impl BomReference {
+    pub fn new<T>(input: T) -> Self
+    where
+        T: ToString,
+    {
+        Self(input.to_string())
+    }
+}
+
+impl AsRef<String> for BomReference {
```

Comment:
I think `AsRef<str>` might be more appropriate. You'd rarely use any `String` specific method considering that you cannot mutate the underlying string through `as_ref` and it will save you at least a deref.

Reply:
Agreed. I'd like to see this addressed before I merge the PR.

Reply:
Implementation deleted, was not used so far.